detox/src/devices/runtime/RuntimeDevice.js

```diff
@@ -119,8 +120,13 @@ class RuntimeDevice {
 
     if (params.delete) {
       await this.terminateApp(bundleId);
-      await this.uninstallApp();
-      await this.installApp();
+
+      if (this._behaviorConfig.optimizeAppInstall) {
```

Comment:
@jonathanmos @asafkorem @d4vidi 

Do we want `delete` to behave ambiguously?

Reply:
This is a good point, and I would say that the user's intent when launching with 'delete: true' is to launch the app with a clean state, so I don't necessarily see resetting the cache as conflicting with that.